### PR TITLE
chore: `pnpm` migration & cleanup

### PR DESCRIPTION
**Pull request by jbroma**
### Summary

This PR deals with many outstanding issues such as:
- troublesome upgrading of dependencies
- heavy duplication of node_modules
- slow installs of node_modules
- addressing ghost dependencies
- CI time reduced by around 50%

Lockfile versions were preserved thanks to [pnpm import](https://pnpm.io/cli/import)

Additional changes associated with this PR:
- [x] replaced all usages of `yarn` with `pnpm` equivalents
- [x] added missing ghost dependencies
- [x] updated some shared devDependencies
- [x] fixed all associated TS/lint/test issues in all packages
- [x] updated TesterApp to RN `0.73.5` for more compatibility with `pnpm`
- [x] moved some reusable `devDependencies` to root (`eslint`, `prettier` etc.)

### Test plan

- [x] - lint works
- [x] - typecheck works
- [x] - test works
- [x] - old arch ios/android TesterApp works
- [x] - new arch ios/android TesterApp works
- [x] - CI works

### Notes

`yarn` with `nodeLinker: pnpm` proved to be unreliable with many extra issues on top of what `pnpm` adds, and `pnpm` has better market penetration.

### CODE REVIEW

File: packages/dev-server/package.json [hunk 1]
```diff
@@ -35,14 +35,14 @@
...
     "build:ts": "tsc --emitDeclarationOnly",
-    "build:ts:watch": "yarn build:ts --watch --preserveWatchOutput",
-    "build": "yarn build:js && yarn build:ts",
+    "build:ts:watch": "pnpm build:ts --watch --preserveWatchOutput",
+    "build": "pnpm build:js && pnpm build:ts",
     "docs": "typedoc",
     "typecheck": "tsc --noEmit",
     "lint": "eslint --ext \".js,.ts\" src",
-    "archive": "yarn build && yarn pack --out repack-dev-server-%v-$(git log -1 --format=\"%h\").tgz"
+    "archive": "pnpm build && pnpm pack"
```
Comment (jbroma) [inline packages/dev-server/package.json:45]: @RafikiTiki not sure if we need custom name there

File: packages/repack/package.json [hunk 2]
```diff
@@ -41,15 +41,15 @@
...
-    "build:ts:watch": "yarn build:ts --watch --preserveWatchOutput",
-    "build": "yarn build:js && yarn build:ts",
+    "build:ts:watch": "pnpm build:ts --watch --preserveWatchOutput",
+    "build": "pnpm build:js && pnpm build:ts",
     "docs": "typedoc && typedoc --options typedoc.client.json",
     "lint": "eslint --ext \".js,.ts\" commands.js client.js src",
     "test": "jest",
     "typecheck": "tsc --noEmit",
-    "archive": "yarn build && yarn pack --out repack-%v-$(git log -1 --format=\"%h\").tgz"
+    "archive": "pnpm build && pnpm pack"
```
Comment (jbroma) [inline packages/repack/package.json:52]: @RafikiTiki not sure if we need custom name there

File: packages/TesterApp/android/gradle.properties [hunk 3]
```diff
@@ -34,7 +34,7 @@ reactNativeArchitectures=armeabi-v7a,arm64-v8a,x86,x86_64
 # your application. You should enable this flag either if you want
 # to write custom TurboModules/Fabric components OR use libraries that
 # are providing them.
-newArchEnabled=false
+newArchEnabled=true
```
Comment (jbroma) [inline packages/TesterApp/android/gradle.properties:37]: @RafikiTiki might be an unneeded change, we got to decide whether we run TesterApp in newarch by default or not

**Review by jbroma — COMMENTED** [review]
pinpointed few places to address for you @RafikiTiki

File: packages/dev-server/package.json (hunk 1 above)
Comment (RafikiTiki) [reply packages/dev-server/package.json:45]: Just checked that and it was used in the early days of Repack for local development/testing against `repack-examples`. We're good to remove that.

File: packages/repack/package.json (hunk 2 above)
Comment (jbroma) [reply packages/repack/package.json:52]: ok brilliant, I couldn't find any reference to that in current codebase as well 👌

**Review by RafikiTiki — APPROVED** [review]
LGTM, tested locally – everything works & clean install in the repo is now a 🚀

File: packages/TesterApp/android/gradle.properties (hunk 3 above)
Comment (RafikiTiki) [reply packages/TesterApp/android/gradle.properties:37]: Let's use newArch to be consistent with iOS

**Pull request merged**